### ISSUE TITLE
feat: alternatieve ML-modellen onderzoeken voor individueel spoor

### ISSUE THREAD

**Issue by Muhammet369** [ed2c/studentprognose#145]
## Context

Het individuele spoor (`-d individual`) gebruikt een vaste twee-staps pipeline:

1. **XGBoost Classifier** (`src/studentprognose/models/xgboost_classifier.py`) — voorspelt per
   vooraanmelder of die zich daadwerkelijk inschrijft (binaire classificatie).
2. **SARIMA** (`src/studentprognose/models/sarima.py:predict_with_sarima_individual`) — aggregeert
   de XGBoost-kansen naar een cumulatieve tijdreeks en extrapoleert tot week 38.

Dit werkt, maar er is nooit onderzocht of andere modellen hier beter presteren. De keuze
voor XGBoost + SARIMA is historisch gegroeid, niet empirisch onderbouwd.

## Probleem

- **XGBoost Classifier** is de enige classificator die is getest. Alternatieven met
  categorische feature-support (CatBoost) of snellere training (LightGBM) zijn niet
  overwogen.
- **SARIMA** met hardcoded ordes is rigide; lichtere of flexibelere tijdreeksmodellen
  (ETS, Prophet) zijn niet vergeleken.
- Er is geen benchmark-framework om modellen eerlijk te vergelijken op dezelfde
  train/test-split.

### Stap 1: Classificatie (vervanging/aanvulling XGBoost Classifier)

| Model | Voordeel | Nadeel |
|---|---|---|
| **LightGBM** | Sneller, native categorische features | Extra dependency |
| **CatBoost** | Native categorische features, minder tuning nodig | Grotere dependency, langzamere training |
| **Logistic Regression** | Interpreteerbaar, snelle baseline | Mist non-lineaire relaties |
| **Random Forest** | Robuust, minder overfitting-risico | Minder nauwkeurig dan boosting bij genoeg data |

### Stap 2: Tijdreeksextrapolatie (vervanging/aanvulling SARIMA)

| Model | Voordeel | Nadeel |
|---|---|---|
| **ETS (Exponential Smoothing)** | Simpeler, minder parameters, stabieler | Geen exogene variabelen |
| **Prophet** | Automatische seizoenaliteit, trend-changepoints | Zware dependency (cmdstanpy) |
| **Holt-Winters** | Licht, goed voor seizoenspatronen | Beperkte flexibiliteit |

### AutoTS als benchmark-tool

[AutoTS](https://github.com/winedarksea/AutoTS) kan ingezet worden om automatisch tientallen tijdreeksmodellen (ARIMA, ETS, Theta, Prophet, etc.) te testen en ranken op dezelfde data. Dit versnelt Fase 1 en 2 aanzienlijk: in plaats van handmatig model voor model te implementeren, laat AutoTS de vergelijking draaien en rapporteert het beste model per reeks.

| Voordeel | Nadeel |
|---|---|
| Test automatisch veel modellen in één run | Zware dependency met veel transitive deps |
| Ingebouwde cross-validatie en metric-vergelijking | Bij korte reeksen (5-8 jaar) mogelijk instabiele modelselectie |
| Ondersteunt ensembles van tijdreeksmodellen | Vervangt alleen het tijdreeksdeel, niet de XGBoost-classificatie |
| Goede exploratietool om snel te zien welke modelfamilie het beste presteert | Black-box: minder controle over waarom een model gekozen wordt |

**Aanbeveling:** gebruik AutoTS in Fase 1 als exploratietool om de meest belovende modelfamilies te identificeren, en implementeer de winnaar(s) vervolgens handmatig in de pipeline voor volledige controle.

### Fase 1: Benchmark-framework

1. Definieer een standaard evaluatie: **TimeSeriesSplit** met vaste train/test-jaren
   (bijv. train t/m jaar N-1, test op jaar N).
2. Meet per model: **MAPE**, **MAE**, **RMSE** en **trainingstijd**.
3. Rapporteer resultaten per opleiding×herkomst×examentype groep.
4. **Draai AutoTS** op de geaggregeerde individuele tijdreeksen om snel de meest belovende modelfamilies te identificeren voordat handmatige experimenten starten.

### Fase 2: Modelexperimenten

5. Train LightGBM en Logistic Regression als classifier-alternatieven naast XGBoost.
6. Train de top-modellen uit de AutoTS-scan als tijdreeks-alternatief naast SARIMA.
7. Vergelijk op de benchmark uit fase 1.

### Fase 3: Integratie

8. Beste model(len) configureerbaar maken via `configuration.json`:
   ```json
   "model_config": {
       "individual_classifier": "xgboost",
       "individual_timeseries": "sarima"
   }
   ```
9. Eventueel: model-stacking (meerdere classifiers combineren).

## Acceptatiecriteria

- [ ] Benchmark-framework met TimeSeriesSplit is beschikbaar
- [ ] AutoTS-scan is gedraaid en resultaten zijn gedocumenteerd
- [ ] Minimaal 2 alternatieve classifiers zijn vergeleken met XGBoost
- [ ] Minimaal 1 alternatief tijdreeksmodel is vergeleken met SARIMA
- [ ] Resultaten zijn gedocumenteerd (MAPE/MAE per model per groep)
- [ ] Beste model is configureerbaar via `configuration.json`

## Referenties

- Huidige pipeline: `IndividualStrategy` in `src/studentprognose/strategies/individual.py`
- Classifier: `src/studentprognose/models/xgboost_classifier.py`
- SARIMA: `src/studentprognose/models/sarima.py`
- AutoTS: https://github.com/winedarksea/AutoTS
- Gerelateerd: #111 (hyperparameter tuning), #147 (cumulatief spoor)

**Comment by Muhammet369** [comment]
## Benchmark resultaten: individueel spoor (`-d i`, classificatie)

Implementatie van het benchmark-framework uit #145 (Fase 1) — classificatiemodellen vergelijken voor het individuele spoor.

### Wat is er gebouwd
- HTML benchmark rapport voor het individuele spoor (`benchmark_individual.html`)
- Interactieve ROC curves met Chart.js: mean curve met confidence band, per-jaar weergave, model toggle
- Drie classifiers vergeleken: XGBoost, Random Forest, Logistic Regression
- TimeSeriesSplit evaluatie: train t/m jaar N-1, test op jaar N

### Resultaten (demo-data)

| Model | Accuracy | AUC-ROC | F1 | Agg MAE | Tijd (s) |
|---|---|---|---|---|---|
| Random Forest | 0.6766 | 0.5016 | 0.8068 | 4.0 | 4.9 |
| XGBoost | 0.6649 | 0.4993 | 0.7969 | 2.2 | 0.2 |
| Logistic Regression | 0.6516 | 0.4985 | 0.7813 | 2.4 | 21.5 |

### ROC curves zijn diagonaal — dat is verwacht met demo-data

Alle drie de modellen scoren een AUC van ~0.50 (random-niveau). Dit komt **niet** door een bug in de implementatie, maar doordat de synthetische demo-data geen signaal bevat: `generate_demo_data.py` kent de inschrijfstatus willekeurig toe (`rng.shuffle`), onafhankelijk van applicant-features zoals school, nationaliteit of vooropleiding.

- **Accuracy ~67%** komt overeen met de base rate (majority class voorspellen)
- **AUC ~0.50** bevestigt dat geen enkel model discriminerend vermogen heeft
- **Alle modellen scoren gelijk** → data-probleem, geen model-probleem

Met echte institutionele data (waar features correleren met inschrijfgedrag) verwachten we dat de ROC curves naar linksboven buigen en de modellen zich van elkaar onderscheiden.

### Status t.o.v. #145

Fase 1 (benchmark-framework + classifier vergelijking) is hiermee afgerond. Fase 2 (AutoTS-scan, alternatieve tijdreeksmodellen) en Fase 3 (model configureerbaar via `configuration.json`) slaan we bewust over — dat wordt overengineering voor de huidige scope. Issue #145 kan hiermee gesloten worden.
